### PR TITLE
Remove FrontEnd CodeCache API methods

### PR DESCRIPTION
**Pull request by 0xdaryl**
* remove FrontEnd unreservedCodeCache()
* remove FrontEnd implementation of needsMethodTrampolines()

### CODE REVIEW

**Comment by 0xdaryl** [comment]
This PR will remain a WIP until eclipse/openj9#672 is merged.

I'd appreciate a review from one of @fjeremic @ymanton @andrewcraik please.

**Comment by 0xdaryl** [comment]
Downstream dependencies are all resolved.  Removing WIP.

**Comment by 0xdaryl** [comment]
@mstoodle @vijaysun-omr : please review and merge.

File: compiler/control/CompileMethod.cpp [hunk 1]
```diff
@@ -449,7 +449,7 @@ compileMethodFromDetails(
    // TR::Compilation. We'll need exceptions working instead of setjmp
    // before we can get working, and we need to make sure the other
    // frontends are properly calling the destructor
-   fe.unreserveCodeCache(compiler.getCurrentCodeCache());
+   TR::CodeCacheManager::instance()->unreserveCodeCache(compiler.getCurrentCodeCache());
```
Comment (mstoodle) [inline compiler/control/CompileMethod.cpp]: copyright update please

Comment (0xdaryl) [reply compiler/control/CompileMethod.cpp]: Fixed

**Pull request merged**